wordpress-mobile/WordPress-iOS
# Login: Nuking awful logic

**Fixes** #8949 

This PR reimplements PR #9051 (by @elibud) in `WordPressAuthenticationManager`, since i **obviously** won the `Who will merge the Breaking PR` race.

@elibud may i trouble you with a hopefully superfun review?

=)

### Testing:

1. Log into a Self Hosted / WordPress.com site. Verify that the Epilogue is displayed.
2. Log into a Self Hosted site. Then sign up from Stats or Notifications. Verify that the sign up epilogue is displayed.
3. Log into a Self Hosted Site. Then log in with an existing account from Stats or Notifications. Verify that the log in epilogue is displayed *if* the new dotcom account had at least one blog.
4. Check that I didn't' break anything else.
